core/src/Dscp/Core/Validation/Educator.hs

```diff
@@ -66,7 +79,15 @@ instance Show BlockValidationFailure where
 instance Buildable BlockValidationFailure where
     build MerkleSignatureMismatch {..} =
       [qc|Merkle tree root signature mismatch. Expected {bvfExpectedSig} got {bvfActualMerkleSig}|]
-    build (SubmissionInvalid x) = build x
+    build (CertificateInvalid x) = build x
+    build (SubmissionInvalid  x) = build x
+
+instance Buildable CertificationValidationFailure where
+    build CertificationSignatureMismatch {..} =
+      mconcat [ [qc|Certification signature mismatch.|]
+              , [qc| Certificate data {cvfCertificationHash} with signature {cvfCertificationSig}|]
+              , [qc| does not correspond to public key {cvfCertificationSigKey}|]
+              ]
```

Comment:
I recently learned about `qm-interpolated-string` package which is like `interpolatedstring-perl6` but allows to deal with line feeds, maybe you'll find it useful.

Reply:
Maybe, but the intent of this commit set differs from upgrading corrent codebase in how the strings are interpolated.